### ISSUE TITLE
More spacious, less information dense redesign of Launchpad

### ISSUE THREAD

**Issue by nteodosio** [canonical/vanilla-framework#4897]
**Describe the bug**

Excessive padding in tables.

**To Reproduce**

Go to any page that contains a table in Launchpad. E.g., https://launchpad.net/ubuntu/+archive/test-rebuild-20230324-lunar/+packages?batch=300&memo=6500&start=6500.

**Expected behavior**

Less padding in the rows.

**Screenshots**

Before:
![lp1](https://github.com/canonical/vanilla-framework/assets/105225958/f5713ed5-424f-46ad-8031-08f95b027977)

After:
![After](https://github.com/canonical/vanilla-framework/assets/105225958/67b1cd19-0671-4b78-9960-a0a3cf2985e7)

**Desktop (please complete the following information):**

- OS: Ubuntu 23.10
- Browser: Firefox, Chromium, Netsurf-gtk
- Version: N/A

**Additional context**

This is of course an opinionated report (calling it a bug would be a stretch) and just one instance of what I'm afraid could be a comprehensive redesign of Launchpad.

The particular change here reported was introduced in [merge 452109](https://code.launchpad.net/~ines-almeida/launchpad/+git/launchpad/+merge/452109) to [conform with Vanilla's guidelines](https://code.launchpad.net/~ines-almeida/launchpad/+git/launchpad/+merge/452109).

I, for one, love Launchpad for the conciseness of its user interface. Where in other sites as Github, Gitlab etc. a screen contains x amount of information, in the same space Launchpad often displays 2x.

That goes for comment sections in bug reports*, merge requests etc., for the user's page, and can be also noted in the parsimony exercised with icons size, which are usually tiny.

Launchpad could therefore be decharacterized if it starts incorporating more spacious variants of design patterns.

I am not a designer and to be clear I'm not trying to impose my views. I'm trying to defend that though there may be general design principles, they might not apply to more specialized software with a specialized audience.

Would proposing a poll, in Discourse for example, be helpful? If the result aligned instead of debunked that essence I tried to capture above, would the changes be reconsidered, or do they need to be carried out for uniformity's sake?

*In contrast, Github also decides to hide some comments from long issues, an epitome. E.g. https://github.com/flatpak/xdg-desktop-portal/pull/705#issuecomment-1686257509.

**Comment by bartaz** [comment]
We can bring this up in the next Vanilla Working Group meeting, although so far neither Vanilla nor Design team (as far as I know) have been involved in Launchpad redesign efforts. So I'm not sure if at this point we are a correct audience for these complains.

Vanilla tables, in current shape, have been used in information-dense apps such as MAAS UI or JAAS dashboard before, so if Launchpad is moving towards using Vanilla it's quite natural to try to use defaults there as well. But as I said, we can bring this up with our designers to see if having denser table option is something to officially approve as part of the design system.

In the meantime feel free to have any additional discussions or pools among interested audience as this will help having more context for devs who are currently applying Vanilla to launchpad, but also for design team.

**Comment by lyubomir-popov** [comment]
We used to have a dense version of the entire framework, and we still have dense versions of some components. 
Table padding used to be affected by that framework wide setting, so table row was 4 instead of 8px IIRC. We do want to maintain a consistent look of all tables, so we should see how Maas, Jaas etc. feel about their current table setup as part of this conversation.

**Comment by nteodosio** [comment]
Thank you for your responses.

I created a [poll in Discourse](https://discourse.ubuntu.com/t/poll-redesign-of-launchpad-elements/39551).

Currently, though not by a far margin, most users hold that change for an improvement.

An additional, more high-level question I asked, "In general, would Launchpad benefit from a redesign with a less compact layout?" is also responded affirmatively by the majority, proving me wrong in my general argument.

> controlled by a setting (the density multiplier)

Allow me a dumb question: This is not a setting available to the user, is it?

**Comment by bartaz** [comment]
> > controlled by a setting (the density multiplier)
> 
> Allow me a dumb question: This is not a setting available to the user, is it?

No. First of all this is not an option available anymore (we simplified spacing since then). And it was a configuration option (SCSS variable) that could be set by developer to simply build ("compile") version of Vanilla with denser vertical spacing. This was only used in early versions of MAAS UI.
